linux/keyman-config/keyman_config/convertico.py

```diff
@@ -33,13 +41,47 @@ def checkandsaveico(icofile):
     im4.save(icofile + ".png")
     os.remove(icofile + ".bmp")
 
+def extractico(kmxfile):
+    """
+    Extract icon file from compiled kmx keyboard
+
+    Args:
+        kmxfile (str): path to kmx file
+    """
+    name, ext = os.path.splitext(kmxfile)
+    icofilename = name+".ico"
+    with open(kmxfile, mode='rb') as file: # b is important -> binary
+        fileContent = file.read()
+
+        kmxstart = struct.unpack_from("<16I", fileContent, 0)
+        if kmxstart[0] != 0x5354584B:
+            logging.debug("bad kmx identifier")
+            return False
+        bitmapOffset = kmxstart[14]
+        bitmapSize = kmxstart[15]
+        logging.debug("bitmap offset is %d", bitmapOffset)
+        logging.debug("bitmap size is %d", bitmapSize)
+        file.seek(bitmapOffset, 0)
+        bitmap = file.read(bitmapSize)
+        if not bitmap:
+            logging.debug("unreadable bitmap in kmx")
+            return False
+        with open(icofilename, mode='wb') as iconfile:
+            iconfile.write(bitmap)
```

Comment:
The icon can be either a .bmp or a .ico. Check the first two bytes of the image data -- if they are `BM` then it's a bitmap file.